### ISSUE TITLE
Fix boundset panic when laying out text

### ISSUE THREAD

**Issue by e-matteson** [justinpombrio/synless#13]
Failing test case:

```
---- test_json_string stdout ----
thread 'test_json_string' panicked at 'No bound fits within given width 2.
Boundset: [***]', pretty/src/layout/boundset.rs:35:17
```

Maybe we should merge the staircase changes first, then debug it.

**Comment by e-matteson** [comment]
Fixed, not PR'ed yet.

**Comment by e-matteson** [comment]
Fixed by #24
